code/modules/mob/living/simple_animal/bot_swarm/swarmer.dm

```diff
@@ -0,0 +1,598 @@
+				if(istype(random_location, /turf/simulated/floor))
+					var/turf/simulated/floor/F = random_location
+					if(F.air)
+						var/datum/gas_mixture/A = F.air
+						if(A.oxygen >= 16 && !A.toxins && A.carbon_dioxide < 10 && !A.trace_gases.len)//Can most things breathe in this location?
+							if((A.temperature > 270) && (A.temperature < 360))//Not too hot, not too cold
+								var/pressure = A.return_pressure()
+								if((pressure > 20) && (pressure < 550))//Account for crushing pressure or vaccuums
+									if(ishuman(target))//If we're getting rid of a human, slap some zipties on them to keep them away from us a little longer
+										var/obj/item/weapon/restraints/handcuffs/cable/zipties/Z = new /obj/item/weapon/restraints/handcuffs/cable/zipties(src)
+										var/mob/living/carbon/human/H = target
+										Z.apply_cuffs(H, src)
+									do_teleport(target, F, 0)
+									playsound(src,'sound/effects/sparks4.ogg',50,1)
+									break
+			return
+
+/mob/living/simple_animal/hostile/swarmer/proc/DismantleMachine(var/obj/machinery/target)
+	do_attack_animation(target)
+	src << "<span class='info'>We begin to dismantle this machine. We will need to be uninterrupted.</span>"
+	var/obj/effect/swarmer/dismantle/D = new /obj/effect/swarmer/dismantle(get_turf(target))
+	D.pixel_x = target.pixel_x
+	D.pixel_y = target.pixel_y
+	D.pixel_z = target.pixel_z
+	if(do_mob(src, target, 100))
+		src << "<span class='info'>Dismantling complete.</span>"
+		var/obj/item/stack/sheet/metal/M = new /obj/item/stack/sheet/metal(target.loc)
+		M.amount = 5
+		if(target.component_parts && target.component_parts.len)
+			for(var/obj/item/I in target.component_parts)
+				I.forceMove(M.loc)
+		var/obj/effect/swarmer/disintegration/N = new /obj/effect/swarmer/disintegration(get_turf(target))
+		N.pixel_x = target.pixel_x
+		N.pixel_y = target.pixel_y
+		N.pixel_z = target.pixel_z
+		target.dropContents()
+		if(istype(target, /obj/machinery/computer))
+			var/obj/machinery/computer/C = target
+			if(C.circuit)
+				var/obj/item/weapon/circuitboard/circuit = text2path(C.circuit)
+				new circuit(get_turf(M))
+		qdel(target)
+
+
+/obj/effect/swarmer //Default swarmer effect object visual feedback
+	name = "swarmer ui"
+	desc = null
+	gender = NEUTER
+	icon = 'icons/mob/swarmer.dmi'
+	icon_state = "ui_light"
+	mouse_opacity = 0
+	layer = 4
+	unacidable = 1
+
+/obj/effect/swarmer/disintegration
+	icon_state = "disintegrate"
+
+/obj/effect/swarmer/disintegration/New()
+	playsound(src.loc, "sparks", 100, 1)
+	spawn(10)
+		qdel(src)
+
+/obj/effect/swarmer/dismantle
+	icon_state = "dismantle"
+
+/obj/effect/swarmer/dismantle/New()
+	spawn(25)
+		qdel(src)
+
+/obj/effect/swarmer/integrate
+	icon_state = "integrate"
+
+/obj/effect/swarmer/integrate/New()
+	spawn(5)
+		qdel(src)
+
+/obj/effect/swarmer/destructible //Default destroyable object for swarmer constructions
+	luminosity = 1
```

Comment:
not sure this var should exist here---@Tigercat2000 would know better than me, but I believe this should be setLight(1).

There's a few other instances of using luminosity here, as well.